python/pyspark/sql/connect/dataframe.py

```diff
@@ -206,7 +208,10 @@ def _repr_html_(self) -> Optional[str]:
 
     @property
     def write(self) -> "DataFrameWriter":
-        return DataFrameWriter(self._plan, self._session)
+        def cb(qe: "ExecutionInfo") -> None:
+            self._execution_info = qe
+
+        return DataFrameWriter(self._plan, self._session, cb)
```

Comment:
Looks like writeStream is not override here. So I imagine streaming query is not supported yet.

In streaming a query could have multiple data frames, what we do in scala is to access it with query.lastExecution

https://github.com/apache/spark/blob/94763438943e21ee8156a4f1a3facddbf8d45797/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala#L192

That's, as it's name, the QueryExecution(`IncrementalExecution`) of the last execution.

We could also add a similar mechanism to StreamingQuery object. This sounds like an interesting followup that im interested in

Reply:
Yes, we should look at streaming as a follow up.